query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -405,7 +405,11 @@ class AvgOverTimeFunction(var sum: Double = Double.NaN, var count: Int = 0) exte
   override def apply(startTimestamp: Long, endTimestamp: Long, window: Window,
                      sampleToEmit: TransientRow,
                      queryConfig: QueryConfig): Unit = {
-    sampleToEmit.setValues(endTimestamp, sum/count)
+    if (count == 0) {
+      sampleToEmit.setValues(endTimestamp, Double.NaN)
+    } else {
+      sampleToEmit.setValues(endTimestamp, sum/count)
+    }
```

Comment:
This should be reverted. There is no divide-by-zero exception when Double is involved. Result is Infinity. 

```
scala> 5d/0d
res0: Double = Infinity
```

Reply:
Hi @vishramachandran What would be the expected behavior? If sum is NaN the division will give NaN irrespective of whether count is 0 or not, but will be Infinity when sum is non NaN and count is 0. The code change above will give NaN in all scenarios.

Reply:
the problem is that it does not match with the Prometheus, the reason, I had to put this in is to match with the Prom behavior. I don't think conceptually -infinity is correct either.